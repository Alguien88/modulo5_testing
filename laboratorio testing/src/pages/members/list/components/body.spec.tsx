import * as React from 'react';
import { shallow, mount } from 'enzyme';
import { Body } from './body';

describe('pages/members/list/components/body tests', () => {
    it('should render as expected when passing required properties', () => {
        // Arrange
        const props = {
            members: [
                {
                    id: 1,
                    name: 'login 1',
                    avatarUrl: 'avatar 1',
                },
                {
                    id: 2,
                    name: 'login 2',
                    avatarUrl: 'avatar 2',
                },
            ],
        };

        // Act
        const component = shallow(
            <Body {...props} />
        );

        // Assert
        expect(component).toMatchSnapshot();
    });

    it('should render as expected when passing required properties with mount', () => {
        // Arrange
        const props = {
            members: [
                {
                    id: 1,
                    name: 'login 1',
                    avatarUrl: 'avatar 1',
                },
                {
                    id: 2,
                    name: 'login 2',
                    avatarUrl: 'avatar 2',
                },
            ],
        };

        // Act
        const component = mount(
            <Body {...props} />
        );

        // Assert
        expect(component).toMatchSnapshot();
    });
});
