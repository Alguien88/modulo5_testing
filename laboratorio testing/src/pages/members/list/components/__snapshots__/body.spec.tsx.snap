// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/members/list/components/body tests should render as expected when passing required properties 1`] = `
<tbody>
  <Component
    key="1"
    member={
      Object {
        "avatarUrl": "avatar 1",
        "id": 1,
        "name": "login 1",
      }
    }
  />
  <Component
    key="2"
    member={
      Object {
        "avatarUrl": "avatar 2",
        "id": 2,
        "name": "login 2",
      }
    }
  />
</tbody>
`;

exports[`pages/members/list/components/body tests should render as expected when passing required properties with mount 1`] = `
<Component
  members={
    Array [
      Object {
        "avatarUrl": "avatar 1",
        "id": 1,
        "name": "login 1",
      },
      Object {
        "avatarUrl": "avatar 2",
        "id": 2,
        "name": "login 2",
      },
    ]
  }
>
  <tbody>
    <Component
      key="1"
      member={
        Object {
          "avatarUrl": "avatar 1",
          "id": 1,
          "name": "login 1",
        }
      }
    >
      <tr>
        <td>
          <img
            src="avatar 1"
            style={
              Object {
                "maxWidth": "10rem",
              }
            }
          />
        </td>
        <td>
          <span>
            1
          </span>
        </td>
        <td>
          <span>
            login 1
          </span>
        </td>
      </tr>
    </Component>
    <Component
      key="2"
      member={
        Object {
          "avatarUrl": "avatar 2",
          "id": 2,
          "name": "login 2",
        }
      }
    >
      <tr>
        <td>
          <img
            src="avatar 2"
            style={
              Object {
                "maxWidth": "10rem",
              }
            }
          />
        </td>
        <td>
          <span>
            2
          </span>
        </td>
        <td>
          <span>
            login 2
          </span>
        </td>
      </tr>
    </Component>
  </tbody>
</Component>
`;
